va_explorer/va_analytics/dash_apps/va_dashboard.py

```diff
@@ -663,65 +690,102 @@ def update_options(search_value, location_json):
     ]
 )
 def filter_data(
-    va_data, selected_json, timeframe="All", search_terms=[], locations=None, location_types=None
-):  
-    filter_df = pd.read_json(va_data)
+    va_data, invalid_va_data, selected_json, timeframe="all", search_terms=[], locations=None, location_types=None
+):
+    if va_data is not None:
+        valid_va_df = pd.read_json(va_data)
+        invalid_va_df = pd.read_json(invalid_va_data)
+        locations = json.loads(locations) if type(locations) is str else locations
+        search_terms = [] if search_terms is None else search_terms
+        location_types = json.loads(location_types) if location_types is not None else location_types
+        disable_timeframe = False
+        
+        # filter valid vas (VAs with COD)
+        valid_filter = _get_filter_dict(valid_va_df, selected_json, timeframe=timeframe,\
+                                          location_types=location_types, search_terms=search_terms,\
+                                          locations=locations)
+        # filter invalid vas (VAs without COD)
+        invalid_filter = _get_filter_dict(invalid_va_df, selected_json, timeframe=timeframe,\
+                                          location_types=location_types, search_terms=search_terms,\
+                                          locations=locations)    
+        # combine filters into one dictionary to share across callbacks
+        combined_filter_dict = {
+            #"granularity": valid_filter["granularity"], # same across both dictionaries
```

Comment:
Is granularity truly no longer needed?

Reply:
Good catch. I was technically expecting it in update_choropleth (filter_dict.get(granularity, DEFAULT_GRANULARITY), but the get statement defaults to initial granularity so it never threw an error. Anyhow, I uncommented it and it's back in the outermost filter dictionary now.